run.py

```diff
@@ -287,45 +358,66 @@ def run_single_test(self, test_index: int, test: dict, runner: str, version: str
         inputs = test['inputs']
         wdl_dir = inputs['dir']
         wdl_input = inputs.get('wdl', f'{wdl_dir}.wdl')  # default wdl name
-        json_input = inputs.get('json', f'{wdl_dir}.json')  # default json name
         abs_wdl_dir = os.path.abspath(wdl_dir)
         if version in WDL_VERSIONS:
             wdl_input = f'{wdl_dir}/{wdl_input}'
         else:
             return {'status': 'FAILED', 'reason': f'WDL version {version} is not supported!'}
+        wdl_file = os.path.abspath(get_wdl_file(wdl_input, abs_wdl_dir, version))
 
-        json_path = f'{wdl_dir}/{json_input}'  # maybe return failing result if no json file found
+        json_input = inputs.get('json')
+        if json_input is None:
+            json_file = None
+        else:
+            json_path = f'{wdl_dir}/{json_input}'
+            json_file = os.path.abspath(json_path)
+
+        json_string = json.dumps(inputs['json_string']) if inputs.get('json_string') else None
+
+        # Only one of json_string or input json file can exist
+        if json_string is not None and json_file is not None:
+            return {'status': 'FAILED', 'reason': f'Config file specifies both a json string and json input file! Only one can be supplied! '
+                                                  f'Check the input section for test id {test["id"]}.'}
 
-        wdl_file = os.path.abspath(get_wdl_file(wdl_input, abs_wdl_dir, version))
-        json_file = os.path.abspath(json_path)
         test_args = args[runner].split(" ") if args[runner] is not None else []
         unique_id = uuid4()
         # deal with jobstore_path argument for toil
         if runner == "toil-wdl-runner" and jobstore_path is not None:
             unique_jobstore_path = os.path.join(jobstore_path, f"wdl-jobstore-{unique_id}")
             test_args.extend(["--jobstore", unique_jobstore_path])
-        outputs = test['outputs']
         results_file = os.path.abspath(f'results-{unique_id}.json')
         wdl_runner = RUNNERS[runner]
         # deal with cromwell arguments to define java system properties
         pre_args = None
         if runner == "cromwell":
             pre_args = args["cromwell_pre_args"]
-        cmd = wdl_runner.format_command(wdl_file, json_file, results_file, test_args, verbose, pre_args)
+
+        # todo: it seems odd that I'm looking for a dependency when the test spec says its supposed to be used to turn failing tests into warnings
```

Comment:
The spec https://github.com/openwdl/wdl-tests/blob/main/docs/Specification.md says if a dependency is detected that a test depends on, but the test fails, it should be turned into a warning.

However, I'm using the dependency field as a way to input more data into the test runner; particularly telling toil to run with docker/singularity as their behavior can differ. I think I did this because one of the unit tests depends on using `findmnt` on `/` and the output differs between singularity and docker.

I changed the comment to be a bit more descriptive.